packages/design-system/src/components/slider/slider.tsx

```diff
@@ -0,0 +1,311 @@
+} from './slider.css';
+
+export const SliderOutputContext =
+  createContext<ContextValue<SliderOutputProps, HTMLOutputElement>>(null);
+
+export const SliderOutput = forwardRef(function SliderOutput(
+  props: SliderOutputProps,
+  ref: ForwardedRef<HTMLOutputElement>,
+) {
+  [props, ref] = useContextProps(props, ref, SliderOutputContext);
+
+  props = useDefaultProps(props, 'SliderOutput');
+
+  const { children: childrenProp, classNames: classNamesProp, ...rest } = props;
+
+  const theme = useTheme();
+
+  const classNames = useMemo(
+    () => mergeClassNames(sliderClassNames, theme.Slider, classNamesProp),
+    [theme.Slider, classNamesProp],
+  );
+
+  const style = useCallback(
+    (renderProps: SliderRenderProps) =>
+      inlineVars(sliderStateVars, renderProps),
+    [],
+  );
+
+  const values = useMemo<
+    [
+      [
+        typeof SliderOutputContext,
+        ContextValue<SliderOutputProps, HTMLOutputElement>,
+      ],
+    ]
+  >(() => [[SliderOutputContext, { classNames }]], [classNames]);
+
+  const children = useCallback(
+    (renderProps: SliderRenderProps) => (
+      <Provider values={values}>
+        {callRenderProps(childrenProp, renderProps)}
+      </Provider>
+    ),
+    [childrenProp, values],
+  );
+
+  return (
+    <RACSliderOutput
+      {...rest}
+      ref={ref}
+      className={classNames?.output?.container}
+      style={style}
+    >
+      {children}
+    </RACSliderOutput>
+  );
+});
+
+/**
+ * SliderTrack must be used as a child of Slider and a parent of SliderThumb
+ */
+
+export const SliderTrackContext =
+  createContext<ContextValue<SliderTrackProps, HTMLDivElement>>(null);
+
+export const SliderTrack = forwardRef(function SliderTrack(
+  props: SliderTrackProps,
+  ref: ForwardedRef<HTMLDivElement>,
+) {
+  [props, ref] = useContextProps(props, ref, SliderTrackContext);
+
+  props = useDefaultProps(props, 'SliderTrack');
+
+  const { children: childrenProp, classNames: classNamesProp, ...rest } = props;
+
+  const theme = useTheme();
+
+  const classNames = useMemo(
+    () => mergeClassNames(sliderClassNames, theme.Slider, classNamesProp),
+    [theme.Slider, classNamesProp],
+  );
+
+  const style = useCallback(
+    (renderProps: SliderTrackRenderProps) =>
+      inlineVars(sliderTrackStateVars, renderProps),
+    [],
+  );
+
+  const values = useMemo<
+    [
+      [
+        typeof SliderThumbContext,
+        ContextValue<SliderThumbProps, HTMLDivElement>,
+      ],
+    ]
+  >(() => [[SliderThumbContext, { classNames }]], [classNames]);
+
+  const children = useCallback(
+    (renderProps: SliderTrackRenderProps) => (
+      <Provider values={values}>
+        {callRenderProps(childrenProp, renderProps)}
+      </Provider>
+    ),
+    [childrenProp, values],
+  );
+
+  return (
+    <RACSliderTrack
+      {...rest}
+      ref={ref}
+      className={classNames?.track?.container}
+      style={style}
+    >
+      {children}
+    </RACSliderTrack>
+  );
+});
+
+/**
```

Comment:
Let's move this down so it's directly next to the SliderThumb component so it gets picked up by intellisense